rdt/transformers/numerical.py

```diff
@@ -156,7 +157,9 @@ def _transform(self, data):
         Returns:
             numpy.ndarray
         """
-        return self.null_transformer.transform(data)
+        if self.nan is not None:
```

Comment:
This does not seem correct.

By applying the NullTransformer *awlays*, we guarantee that no Null values make their way to the output *even if there were no nulls in the fit data*, which stops being true after this change.
On the other hand, bear in mind that `self.nan` being `None` does not mean "do not replace nulls", but rather "choose what you use to replace nulls depending on the dtype of the input data"